### ISSUE TITLE
API not working anymore

### ISSUE THREAD

**Issue by Shadorc** [alfg/overwatch-api#80]
Hi,

Blizzard has probably pushed some breaking changes to their Career Overview page because overwatch-api is returning null for all fields or maybe the API is down.

Example:
https://playoverwatch.com/en-gb/career/pc/Shadorc-2503
https://owapi.io/profile/pc/global/Shadorc-2503

**Comment by alfg** [comment]
Thanks for reporting this. I'll check into this today.

**Comment by alfg** [comment]
Running locally seems to be fine. However, I think the deployed server is getting throttled from the playoverwatch website from too many requests.

```
curl -i https://playoverwatch.com/en-gb/career/pc/Shadorc-2503
HTTP/2 403
server: CloudFront
date: Sun, 05 Apr 2020 19:47:15 GMT
content-type: text/html
content-length: 919
x-cache: Error from cloudfront
via: 1.1 c2354f8128e4d7b61eb5ba62ca8e4aea.cloudfront.net (CloudFront)
x-amz-cf-pop: SFO20-C1
x-amz-cf-id: EopNHAF6itkgnzWB74olDGUfDXX693GlKVQ0EDXS--27gH285_2Msw==

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<HTML><HEAD><META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=iso-8859-1">
<TITLE>ERROR: The request could not be satisfied</TITLE>
</HEAD><BODY>
<H1>403 ERROR</H1>
<H2>The request could not be satisfied.</H2>
<HR noshade size="1px">
Request blocked.
We can't connect to the server for this app or website at this time. There might be too much traffic or a configuration error. Try again later, or contact the app or website owner.
<BR clear="all">
If you provide content to customers through CloudFront, you can find steps to troubleshoot and help prevent this error by reviewing the CloudFront documentation.
<BR clear="all">
<HR noshade size="1px">
<PRE>
Generated by cloudfront (CloudFront)
Request ID: EopNHAF6itkgnzWB74olDGUfDXX693GlKVQ0EDXS--27gH285_2Msw==
</PRE>
<ADDRESS>
</ADDRESS>
```

I may need to add more API rate limiting on my end to prevent this or spread the load across more servers.

**Comment by Shadorc** [comment]
Ok, let me know if there is any news!

**Comment by alfg** [comment]
It should be back up (for now). However, I'm still working on rate limits and possibly longer caching to prevent getting throttled.

**Comment by Shadorc** [comment]
That's a nice idea :) Good luck!
And thanks for your awesome API

**Comment by alfg** [comment]
OK, I've added #81 to support configured caching TTL. I've updated `owapi.io` to 15 minute TTL and updated the rate limiting rules a bit. We'll see how long this works for. 😆 Hopefully this will help prevent any throttle limits on the playoverwatch page.

However, I'm sure this is going to increase memory usage, so I'll need to look into caching into a store, such as redis. I've also increased memory a bit on the server. 

Thanks again for reporting this, I appreciate it! :)
